libraries/ESP8266WiFi/src/WiFiClientBearSSL.cpp

```diff
@@ -0,0 +1,1203 @@
+bool WiFiClientBearSSL::_connectSSLServerRSA(const BearSSLX509List *chain,
+                                             const BearSSLPrivateKey *sk,
+                                             const BearSSLX509List *client_CA_ta)
+{
+  _freeSSL();
+  _oom_err = false;
+  _sc_svr = std::make_shared<br_ssl_server_context>();
+  _eng = &_sc_svr->eng; // Allocation/deallocation taken care of by the _sc shared_ptr
+  _iobuf_in = std::shared_ptr<unsigned char>(new unsigned char[_iobuf_in_size], std::default_delete<unsigned char[]>());
+  _iobuf_out = std::shared_ptr<unsigned char>(new unsigned char[_iobuf_out_size], std::default_delete<unsigned char[]>());
+
+  if (!_sc_svr || !_iobuf_in || !_iobuf_out) {
+    _freeSSL();
+    _oom_err = true;
+    return false;
+  }
+
+  br_ssl_server_init_full_rsa(_sc_svr.get(), chain?chain->getX509Certs():nullptr, chain?chain->getCount():0, sk?sk->getRSA():nullptr);
+  br_ssl_engine_set_buffers_bidi(_eng, _iobuf_in.get(), _iobuf_in_size, _iobuf_out.get(), _iobuf_out_size);
+  if (client_CA_ta && !_installServerX509Validator(client_CA_ta)) {
+    return false;
+  }
+  if (!br_ssl_server_reset(_sc_svr.get())) {
+    _freeSSL();
+    return false;
+  }
+
+  return _wait_for_handshake();
+}
+
+// Called by WiFiServerBearSSL when an elliptic curve cert/key is specified.
+bool WiFiClientBearSSL::_connectSSLServerEC(const BearSSLX509List *chain,
+                                            unsigned cert_issuer_key_type, const BearSSLPrivateKey *sk,
+                                            const BearSSLX509List *client_CA_ta)
+{
+  _freeSSL();
+  _oom_err = false;
+  _sc_svr = std::make_shared<br_ssl_server_context>();
+  _eng = &_sc_svr->eng; // Allocation/deallocation taken care of by the _sc shared_ptr
+  _iobuf_in = std::shared_ptr<unsigned char>(new unsigned char[_iobuf_in_size], std::default_delete<unsigned char[]>());
+  _iobuf_out = std::shared_ptr<unsigned char>(new unsigned char[_iobuf_out_size], std::default_delete<unsigned char[]>());
+
+  if (!_sc_svr || !_iobuf_in || !_iobuf_out) {
+    _freeSSL();
+    _oom_err = true;
+    return false;
+  }
+
+  br_ssl_server_init_full_ec(_sc_svr.get(), chain?chain->getX509Certs():nullptr, chain?chain->getCount():0,
+                             cert_issuer_key_type, sk?sk->getEC():nullptr);
+  br_ssl_engine_set_buffers_bidi(_eng, _iobuf_in.get(), _iobuf_in_size, _iobuf_out.get(), _iobuf_out_size);
+  if (client_CA_ta && !_installServerX509Validator(client_CA_ta)) {
+    return false;
+  }
+  if (!br_ssl_server_reset(_sc_svr.get())) {
+    _freeSSL();
+    return false;
+  }
+
+  return _wait_for_handshake();
+}
+
+// Returns an error ID and possibly a string (if dest != null) of the last
+// BearSSL reported error.
+int WiFiClientBearSSL::getLastSSLError(char *dest, size_t len)
+{
+  int err = 0;
+  const char *t = PSTR("OK");
+  if (_sc || _sc_svr) err = br_ssl_engine_last_error(_eng);
+  if (_oom_err) err = -1000;
+  switch (err) {
```

Comment:
this switch of 1-liners makes my ears pop, but I understand the rationale: it's almost table-like, and formatting it would make it huge.
But: no default (error handling)

Reply:
Good catch.